automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/apps/AppCreationSpec.scala

```diff
@@ -65,17 +66,24 @@ class AppCreationSpec extends GPAllocFixtureSpec with LeonardoTestUtils with GPA
           _ = getAppResponse.status should (be(AppStatus.Deleting) or be(AppStatus.Predeleting))
 
           // Verify the app eventually becomes Deleted
+          // Don't fail the test if the deletion times out because the Galaxy pre-delete job can sporadically fail.
+          // See https://broadworkbench.atlassian.net/browse/IA-2471
           listApps = LeonardoApiClient.listApps(googleProject, true)
-          monitorDeleteResult <- streamUntilDoneOrTimeout(
+          implicit0(deletedDoneCheckable: DoneCheckable[List[ListAppResponse]]) = appDeleted(appName)
+          monitorDeleteResult <- streamFUntilDone(
             listApps,
             120,
-            10 seconds,
-            s"AppCreationSpec: app ${googleProject.value}/${appName.value} did not finish deleting after 20 minutes"
-          )(implicitly, implicitly, appDeleted(appName))
-          _ <- loggerIO.info(
-            s"AppCreationSpec: app ${googleProject.value}/${appName.value} delete result: $monitorDeleteResult"
-          )
-
+            10 seconds
+          ).compile.lastOrError
+          _ <- if (!deletedDoneCheckable.isDone(monitorDeleteResult)) {
```

Comment:
maybe we can add `result` to `StreamTimeoutError` so that we can support this case

Reply:
That's a good idea